src/logfile_values_experimental.py

```diff
@@ -0,0 +1,221 @@
+
+"""
+import re
+import time
+import os
+import gzip
+import logging
+
+from argparse import ArgumentParser, ArgumentTypeError
+from datetime import timedelta
+
+
+class Metric:
+    def __init__(self, arg):
+        if ':' not in arg:
+            raise ArgumentTypeError('Argument must have ":"')
+        parts = arg.split(':')
+        if len(parts) != 4 and len(parts) != 2:
+            raise ValueError('Wring number of options')
+
+        # Accepts 2 or 4 options. If 2 then function and period are NULL
+        parts += [None] * (4 - len(parts))
+        self.name, column, function, period = parts
+        if period:
+            pattern = re.compile("^\d+[A-Za-z]$")
+            if not pattern.match(period):
+                raise ArgumentTypeError('Period must have number and unit')
+        if not column.isdecimal():
+            raise ArgumentTypeError('Column must be a number')
+        self.column = int(column)
+        self.function = function
+        self.period = period
+        # Here is container for metric values
+        self.values = []
+        self.last_value = 0
+
+    def get_timeshift(self):
+        if self.period:
+            value = int(self.period[:-1])
+            unit = self.period[-1].lower()
+            now = int(time.time())
+            if unit == 's':
+                return timedelta(seconds=value).total_seconds()
+            elif unit == 'm':
+                return timedelta(minutes=value).total_seconds()
+            elif unit == 'h':
+                return timedelta(hours=value).total_seconds()
+            elif unit == 'd':
+                return timedelta(days=value).total_seconds()
+        else:
+            return 0
+
+    def get_median(self):
+        l = sorted(self.values)
+        i = len(l)
+        if not i % 2:
+            return (l[(i // 2) - 1] + l[i // 2]) / 2
+        return int(l[i // 2])
+
+    def get_sum(self):
+        return int(sum(self.values))
+
+    def get_count(self, v=0):
+        return int(sum(1 for x in self.values if x > v))
+
+    def get_mean(self):
+        return int(sum(self.values) / len(self.values))
+
+    def get_min(self):
+        return int(min(self.values))
+
+    def get_max(self):
+        return int(max(self.values))
+
+    def get_last_value(self):
+        return int(self.last_value)
+
+    def get_metric_value(self):
+        f = self.function
+        if f == 'mean':
+            return self.get_mean()
+        elif f == 'median':
+            return self.get_median()
+        elif f == 'sum':
+            return self.get_sum()
+        elif f == 'min':
+            return self.get_min()
+        elif f == 'max':
+            return self.get_max()
+        elif f == 'count':
+            return self.get_count()
+        elif f == 'last' or not f:
+            return self.get_last_value()
+        else:
+            raise ArgumentTypeError(
+                'Wrong function. Possible functions: blablabla')
```

Comment:
We can make this selection more elegant by something like `return getattr(self, 'get_' + self.function)`.